File: Mixter.Domain/Core/Messages/Events/MessageRepublished.cs
using Mixter.Domain.Identity;

namespace Mixter.Domain.Core.Messages.Events
{
    public struct MessageRepublished : IDomainEvent
    {
        public MessageId Id { get; private set; }

        public UserId Republisher { get; private set; }

        public MessageRepublished(MessageId id, UserId republisher)
            : this()
        {
            Id = id;
            Republisher = republisher;
        }

        public object GetAggregateId()
        {
            return Id;
        }
    }
}

File: app/Domain/Messages/MessageRepublished.php
<?php

namespace App\Domain\Messages;

use App\Domain\Identity\UserId;
use App\Domain\IDomainEvent;

class MessageRepublished implements IDomainEvent
{
    /** @var MessageId */
    private $messageId;

    /** @var UserId */
    private $republisherId;

    public function __construct(MessageId $messageId, UserId $republisherId)
    {
        $this->messageId = $messageId;
        $this->republisherId = $republisherId;
    }

    /**
     * @return MessageId
     */
    public function getMessageId()
    {
        return $this->messageId;
    }

    /**
     * @return UserId
     */
    public function getRepublisherId()
    {
        return $this->republisherId;
    }

    /**
     * @return string
     */
    public function getAggregateId()
    {
        return $this->getMessageId()->getId();
    }
}